packages/teleport/src/cluster/components/Users/UserDialog/UserDialog.tsx

```diff
@@ -0,0 +1,204 @@
+    attemptActions.do(() =>
+      onCreateInvite(username, roles).then(token => {
+        setToken(token);
+      })
+    );
+  }
+
+  return (
+    <Validation>
+      {({ validator }) => (
+        <Dialog
+          dialogCss={() => ({ maxWidth: '500px', width: '100%' })}
+          disableEscapeKeyDown={false}
+          onClose={onClose}
+          open={true}
+        >
+          <DialogHeader>
+            {attempt.isSuccess && <DialogTitle>Share Invite Link</DialogTitle>}
+            {!attempt.isSuccess && (
+              <DialogTitle>
+                {dialog.state === 'create'
+                  ? 'Create User Invite Link'
+                  : 'Viewing'}
+              </DialogTitle>
+            )}
+          </DialogHeader>
+          <DialogContent>
+            {attempt.isFailed && (
+              <Alert kind="danger" children={attempt.message} />
+            )}
+            {attempt.isSuccess && <TokenContent token={token} />}
+            {!attempt.isSuccess && (
+              <>
+                <FieldInput
+                  label="User Name"
+                  rule={requiredField('User name is required')}
+                  autoFocus
+                  value={username}
+                  onChange={e => setUsername(e.target.value)}
+                  readonly={dialog.state === 'view' ? true : false}
+                />
+                <FieldSelect
+                  label="Assign Role/s"
+                  rule={requiredField('At least one role is required')}
+                  maxMenuHeight={110}
+                  placeholder="Click to select a role"
+                  isSearchable
+                  isMulti
+                  isSimpleValue
+                  clearable={false}
+                  value={selectedRoles}
+                  onChange={values => setSelectedRoles(values as Option[])}
+                  options={selectOptions}
+                  isDisabled={dialog.state === 'view' ? true : false}
+                />
+              </>
+            )}
+          </DialogContent>
+          <DialogFooter>
+            {(attempt.isSuccess || dialog.state === 'view') && (
+              <ButtonSecondary onClick={onClose}>Close</ButtonSecondary>
+            )}
+            {!attempt.isSuccess && dialog.state === 'create' && (
+              <>
+                <ButtonPrimary
+                  mr="3"
+                  disabled={attempt.isProcessing}
+                  onClick={() => handleOnCreate(validator)}
+                >
+                  {'Create User Invite Link'}
+                </ButtonPrimary>
+                <ButtonSecondary
+                  disabled={attempt.isProcessing}
+                  onClick={onClose}
+                >
+                  Cancel
+                </ButtonSecondary>
+              </>
+            )}
+          </DialogFooter>
+        </Dialog>
+      )}
+    </Validation>
+  );
+}
+
+const TokenContent = ({ token }: TokenContentProps) => {
```

Comment:
`TokenContent` -> `InviteLinkInfo`

Also, since it has a very simple pros, you can incline type definition. 

({ token: } : { token: string}) => 

or even shorter

({ token = "" }  =>

Default value (which is an empty string) tells typescript about expected type (string).

Reply:
The token object contains several props:
```
export interface ResetToken {
  url: string;
  expires: string;
  username: string;
}
```

But I simplified: 
`const TokenContent = ({ token }: TokenContentProps) => {`

to
`const InviteLinkInfo = ({ token }: { token: ResetToken }) => {`